jupyter_server_proxy/handlers.py

```diff
@@ -363,7 +364,85 @@ async def proxy(self, host, port, proxied_path):
                 body = b""
             else:
                 body = None
+        accept_type = self.request.headers.get('Accept')
+        if accept_type == 'text/event-stream':
+            return await self._proxy_progressive(host, port, proxied_path, body)
+        else:
+            return await self._proxy_normal(host, port, proxied_path, body)
+    
+    async def _proxy_progressive(self, host, port, proxied_path, body):
+        # Proxy in progressive flush mode, whenever chunks are received. Potentially slower but get results quicker for voila
+
+        client = httpclient.AsyncHTTPClient()
+
+        # Set up handlers so we can progressively flush result
+
+        headers_raw = []
+
+        def dump_headers(headers_raw):
```

Comment:
let's move this function outside `_proxy_progressive`. `headers_raw` as used inside this function is an arg, while there's also a local variable with the same name. Moving it outside cleans this up a little.